_posts/2020-11-09-introduction-to-github.md

```diff
@@ -0,0 +1,69 @@
+  * **Person 1:** Creates working code for what the project calls for and pushes this code to GitHub
+  * **Person 2:** Fork the code onto their GitHub account and copy the code onto their machine. Make changes to the code. Push code to their GitHub. Open a pull request from the original code.
+  * **Person 1:** Look at Person 2’s pull request. If the code is good, approve the pull request and merge the changes into the main project
+
+This is just the very basics of what two people can do when working collaboratively through GitHub *(there is a lot of jargon in there that I will cover in the Version Control section of this post)*. But the key takeaway is that GitHub allows people to work collaboratively in a way that allows people to do their own work and not mess with each other’s progress.
+
+## The Importance of Version Control
+
+![version control]({{ site.url }}/assets/version_control.png)
+
+It is very important that you keep track of what you’ve done and what works when writing code. Version control goes hand-in-hand with collaborative coding. As previously mentioned, GitHub is great for collaborative projects because of its version control ability. If Person 2 were to work directly on the code that Person 1 was developing, that would break the code and ruin the whole project.
+
+GitHub allows for version control in several ways, some of which are detailed below. For continuity’s sake, I will be referencing the two-person project I detailed in the previous section:
+  * **The ability to fork a repository:** Person 2 can go to person 1’s GitHub profile and find the repository that has their project in it. They can then fork this repository to their GitHub account.
+    * This basically means that they copy the project to their GitHub account so they can use it as their own and not mess with what Person 1 is doing.
+    * Additionally, you have the ability to create new branches in a project. This acts as an effective form of version control as well. A new branch essentially a copy of the original code but it will not effect any of the code on the main branch.
+      * Some command when working with branches are as follows:
+        * Creating a new branch: `git checkout -b <name-of-new-branch>`
+        * Switching to an existing branch: `git checkout <name-of-existing-branch>`
+        * Preview all existing branches: `git branch`
+
+
+  * **Pushing Code to GitHub:** if you are working on the command line this can be done through the command ‘git push’ followed by the name of the file that you would like to push to GitHub. This allows Person 2 to take whatever they were working on and put it onto GitHub for Person 1 to see.
```

Comment:
Not precisely correct. It should be `git push` followed by the name of the remote (often `origin`) and then the name of the branch on the remote to push to (`master` or `main`). Git pushes and pulls commits, not files.